mlrun/runtimes/local.py

```diff
@@ -362,17 +363,33 @@ def run_exec(cmd, args, env=None, cwd=None):
     if env and "SYSTEMROOT" in os.environ:
         env["SYSTEMROOT"] = os.environ["SYSTEMROOT"]
     print("running:", cmd)
-    process = Popen(cmd, stdout=PIPE, stderr=PIPE, env=os.environ, cwd=cwd)
+    process = Popen(
+        cmd, stdout=PIPE, stderr=PIPE, env=os.environ, cwd=cwd, universal_newlines=True
+    )
+
+    def read_stderr(stderr):
+        while True:
+            nextline = process.stderr.readline()
+            if not nextline:
+                break
+            stderr[0] += nextline
+
+    # ML-3710. We must read stderr in a separate thread to drain the stderr pipe so that the spawned process won't
+    # hang if it tries to write more to stderr than the buffer size (default of approx 8kb).
+    stderr = [""]
```

Comment:
why stderr has to be a list and not an str?

Reply:
To get a reference type for inter-thread communication, but now using `StringIO` so list is no longer necessary.